### ISSUE TITLE
Modeling Tabular Data using Conditional GAN

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#74]
## Abstract
* 종종 continuous columns는 multiple modes 가지고, discrete columns는 imbalance해서 modeling을 어렵게 함.
* 본 논문에서 제안하는 CTGAN은 conditional generator를 써서 이러한 문제를 다룸.

## Introduction
* GAN 모델들은 modeling distributions에서 greater flexibility를 제공함.
→ 이러한 GAN 모델들을 평가하기 위해, real datasets 그룹을 씀.
* 비교 목적으로 Bayesian networks를 쓴 2개의 baseline methods 만듦.
* modeling tabular data poses unique challenges for GANs
→fall short of the baseline methods on a number of metrics
* conditional tabular GAN (CTGAN) 제안.
* 아래 new techniqes를 도입함.
  * augmenting the training procedure with mode-specific normalization
  * architectural changes
  * addressing data imbalance by **_employing a conditional generator and training-by-sampling_**
* variational autoencoder (VAE) for mixed-type tabular data generation 적용 →TVAE라고 부름.

## Related Work
* medical dataset 관련 연구가 꽤 있어서 나중에 꼭 논문 찾아서 읽어보기!
* GAN을 continuous time series medical recods generating에 씀.
→ Generative adversarial networks for electronic health records: A framework for exploring and evaluating methods
for predicting drug-induced laboratory test trajectories (논문 제목)
* medGAN
→ Generating multi-label discrete patient records using generative adversarial networks
* ehrGAN
→Boosting deep learning risk prediction with generative adversarial networks for electronic health records
* 나머지는 medical이 아니라 패스..

#### 일단 notation 정리부터
* data synthesizer G
* table T
* synthetic table T_syn
* continuous columns N_c = {C_1, C_2, ...}
* discrete columns N_d = {D_1, D_2,...}
* 위에 두 컬럼에서 각 컬럼은 random variable로 여겨짐.
* joint distribution r_j = {c_1,j,...,C_N_c,j, d_1,j,...,d_N_d,j} , j는 {1,...,n} 중 하나
* T는 T_train과 T_test로 나뉨.
#### notation 끝
* efficacy of a generator along 2 axes는 아래 2가지로 평가함.
  * Likelihood fitness
  * Machine learning efficacy
 * challenges
   * Mixed data types
   * Non-Gaussian distributions
   * Multimodal distributions
   * Learning from sparse one-hot-encoded vectors
   * Highly imbalanced categorical columns
 
## CTGAN Model
* imbalanced discrete columns를 다루기 위해 conditional generator와 training-by-sampling을 디자인함.
* high-quality model을 학습시키기 위해 fc networks와 several recent techniques를 사용함.
#### notation 추가
<image src="https://user-images.githubusercontent.com/67627471/129709935-1391545f-d23e-4768-a0f7-988397f63a08.png" width="70%" height="70%" />

#### Mode-specific Normalization
* complicated distribution이 있는 columns를 다루기 위함.
* mode-specific normalization 예시.
<image src="https://user-images.githubusercontent.com/67627471/129710328-8032a054-9667-4fe7-a8fd-5e70085abe13.png" width="60%" height="60%" />

* each value는 mode를 나타내는 one-hot vector로 표현됨.
* mode-specific normalization 3단계
  * for each continuous column C_i, VGM (variational Gaussian mixture model) 사용.
  → number of modes m_i를 평가하고, Gaussian mixture를 fit하기 위해서.
  → 예를 들어, VGM은 three modes (m_i = 3)을 찾는다. (위에 첨부한 이미지에서)
  * for each value c_i,j in C_i, compute the probability of c_i,j coming from each mode.
  * sample one mode from given the probability density & use the sampled mode to normalize the value.
* 식으로 나타내면,
<image src="https://user-images.githubusercontent.com/67627471/129711622-df93e587-acb8-4dff-ba59-c75e6935a41f.png" width="80%" height="80%" />

#### Conditional Generator and Training-by-Sampling
* Discriminator와 Critic neural networks를 같이 학습시켜서 deterministic transformation을 얻음.
* 학습 데이터가 랜덤하게 샘플되면 class imbalance 문제 발생함.
* 그래서 Conditional generator 위에 GAN을 올려서 씀 -> Conditional GAN
* CTGAN 모델 구조.
<image src="https://user-images.githubusercontent.com/67627471/129712247-4e1afca3-93c2-463a-bf6f-547d9e09b628.png" width="70%" height="70%" />

* condition을 mask vector term으로 표현.
<image src="https://user-images.githubusercontent.com/67627471/129712517-177c32e5-8208-442d-9a7e-c5c8c76ebd58.png" width="30%" height="30%" />

* Generator loss : the generator leans to make an exact copy of the given m_i* into ^d_i*
* Training-by-sampling
<image src="https://user-images.githubusercontent.com/67627471/129712872-f10119cb-9621-4105-9072-2e6ff22e04db.png" width="70%" height="70%" />

#### network structure
<image src="https://user-images.githubusercontent.com/67627471/129713239-e6e497f7-8e76-4db6-8795-6ae02626efbe.png" width="70%" height="70%" />

#### TVAE Model
<image src="https://user-images.githubusercontent.com/67627471/129713434-5e85dd59-0a4b-4b2d-aa08-5229d8245aa8.png" width="70%" height="70%" />

## Evaluation & Results
* Evaluation framework (simulated 데이터와 실제 데이터)
<image src="https://user-images.githubusercontent.com/67627471/129713595-027c40b0-adac-4c68-97c7-6c9e2516cbfd.png" width="80%" height="80%" />

* Benchmark 결과
<image src="https://user-images.githubusercontent.com/67627471/129713770-a6acdfb9-262b-4196-a2a4-2b52813d69d1.png" width="50%" height="50%" />

* with mode-specific normalization, 2D continuous datasets에서 성능 우수함.
* Baysian 네트워크들보다 distributions를 더 잘 학습함.
* conditional generator와 training-by-sampling이 imbalance training data 문제를 잘 극복함.

##
* 꼭 필요한 내용이라 자세히 읽었더니 너무 길어졌네요..
